sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -67,89 +68,116 @@ import org.apache.spark.util.CircularBuffer
 object QueryExecutionErrors {
 
   def columnChangeUnsupportedError(): Throwable = {
-    new UnsupportedOperationException("Please add an implementation for a column change here")
+    new UnsupportedOperationException(
```

Comment:
Can you add a new exception type that mixes `UnsupportedOperationException` with `SparkThrowable`? It'd look like [`SparkArithmeticException`](https://github.com/apache/spark/blob/f90eb6a5db0778fd18b0b544f93eac3103bbf03b/core/src/main/scala/org/apache/spark/SparkException.scala#L75). Then we can catch these exceptions and use their error classes/SQLSTATEs.

Reply:
Of course, I'll do it. Thank you.

Reply:
@karenfeng I also add `SparkUnsupportedOperationException`, `SparkIllegalStateException`,`SparkNumberFormatException`,`SparkIllegalArgumentException`,`SparkArrayIndexOutOfBoundsException`,`SparkNoSuchElementException` that with `SparkThrowable`.